packages/babel-helpers/src/helpers.js

```diff
@@ -442,6 +442,23 @@ helpers.specRequireInterop = template(`
   })
 `);
 
+helpers.specImportCheck = template(`
+  (function (module, imports) {
+    if (!module.__esModule) throw new Error("Only ES modules can be checked");
+    const realImports = Object.keys(module);
+    const invalid = imports.filter(function (i) { return realImports.indexOf(i) < 0; });
+    if (invalid.length > 0) {
+      const error = new Error(
+        "Unknown exports" + (invalid.length > 1 ? "s " : " ") +
+        invalid.map(function (i) { return "'" + i + "'"; }).join(', ') +
```

Comment:
not sure if this is better or not, but `"'" + invalid.join("', '") + "' imported"` avoids the map (since you know the array is nonempty)